src/utilities/district.ts

```diff
@@ -26,40 +32,31 @@ export function getQuotientsToVulnerableSeatMap(districtResult: DistrictResult):
     return partyCodeToQuotientMap;
 }
 
-/**
- * Takes all district results and returns the most vulnerable district based on
- * quotient with its winner and runner up, and votes needed to win.
- *
- * @param districtResult result to find vulnerable seat by quotient for
- *
- * @returns vulnerable seat by quotient
- */
-export function getMostVulnerableSeatByQuotient(districtResults: DistrictResult[]) {
-    const vulnerableDistrictSeats: VulnerableDistrictSeat[] = [];
-    districtResults.forEach((districtResult) => {
-        if (districtResult.districtSeats > 0) {
-            vulnerableDistrictSeats.push({
-                ...getVulnerableSeatByQuotient(districtResult),
-                district: districtResult.name,
-            });
-        }
-    });
-    return vulnerableDistrictSeats.sort((a, b) => (a.moreVotesToWin >= b.moreVotesToWin ? 1 : -1))[0];
-}
-
 /**
  * Takes a district result and compares the winner to its quotient runner up.
  *
  * @param districtResult result to find vulnerable seat by quotient for
  *
  * @returns vulnerable seat by quotient
  */
-export function getVulnerableSeatByQuotient(districtResult: DistrictResult): VulnerableDistrictSeat {
+export function getVulnerableSeatByQuotient(
+    districtResult: DistrictResult,
+    partyResultMap: _.Dictionary<PartyResult>,
+    districtThreshold: number
+): VulnerableDistrictSeat {
     const lastSeat = districtResult.districtSeatResult[districtResult.districtSeatResult.length - 1];
     const winner = lastSeat.partyResults.find((pr) => pr.partyCode === lastSeat.winner)!;
     const lastSeatByQuotient = lastSeat.partyResults.sort((a, b) => (a.quotient <= b.quotient ? 1 : -1));
-    const runnerUp = lastSeatByQuotient[1];
-    const moreVotesToWin = Math.floor(winner.quotient * runnerUp.denominator) - runnerUp.votes + 1;
+    const filtered = lastSeatByQuotient.filter(
```

Comment:
Not a good variable name. It's clear it is filtered by the function called -- it is less clear exactly what this is representing.

Reply:
Fixed.